packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts

```diff
@@ -0,0 +1,122 @@
+import * as crypto from 'crypto';
+import { ISecurityGroup, ISubnet } from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { StreamEventSource, StreamEventSourceProps } from './stream';
+
+/**
+ * Properties for a Kafka event source
+ */
+export interface KafkaEventSourceProps extends StreamEventSourceProps {
+  /**
+   * the Kafka topic to subscribe to
+   */
+  readonly topic: string,
+  /**
+   * the secret with the Kafka credentials, see https://docs.aws.amazon.com/msk/latest/developerguide/msk-password.html for details
+   */
+  readonly secret: secretsmanager.ISecret
+}
+
+/**
+ * Properties for a MSK event source
+ */
+export interface ManagedKafkaEventSourceProps extends KafkaEventSourceProps {
+  /**
+   * the ARN of the MSK cluster
+   */
+  readonly clusterArn: string
+}
+
+/**
+ * Properties for a self managed Kafka cluster event source.
+ * If your Kafka cluster is only reachable via VPC make sure to configure it.
+ */
+export interface SelfManagedKafkaEventSourceProps extends KafkaEventSourceProps {
+  /**
+   * list of Kafka brokers
+   */
+  readonly bootstrapServers: string[]
+
+  /**
+   * If your Kafka brokers are only reachable via VPC, provide the subnets here
+   *
+   * @default - none
+   */
+  readonly subnets?: ISubnet[],
+
+  /**
+   * If your Kafka brokers are only reachable via VPC, provide the security group here
+   *
+   * @default - none
+   */
+  readonly securityGroup?: ISecurityGroup
+}
+
+/**
+ * Use a MSK cluster as a streaming source for AWS Lambda
+ */
+export class ManagedKafkaEventSource extends StreamEventSource<ManagedKafkaEventSourceProps> {
+
+  constructor(props: ManagedKafkaEventSourceProps) {
+    super(props);
+  }
+
+  public bind(target: lambda.IFunction) {
+    target.addEventSourceMapping(
+      `KafkaEventSource:${this.props.clusterArn.split('/').slice(-1)[0]}${this.props.topic}`,
+      this.enrichMappingOptions({
+        eventSourceArn: this.props.clusterArn,
+        startingPosition: this.props.startingPosition,
+        sourceAccessConfigurations: [{ type: 'SASL_SCRAM_512_AUTH', uri: this.props.secret.secretArn }],
```

Comment:
Looking at the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-eventsourcemapping-sourceaccessconfiguration.html#cfn-lambda-eventsourcemapping-sourceaccessconfiguration-type), this is supported only for self managed Kafka clusters, no?

Reply:
It's supported for both types. I actually [wrote](https://github.com/awsdocs/aws-cloudformation-user-guide/commit/7e7f10df359dccf4fe90d51bd41306f16ed0e510#diff-c73ac817fec91622c6b044a71470d851a321405d3980f83f789742d13d870cc9) those docs 🙈 . But somebody added the "self managed" to the published version. We have that exact setup running for a customer.

Reply:
For a managed cluster:
> Amazon MSK only supports SCRAM-SHA-512 authentication.


See https://docs.aws.amazon.com/msk/latest/developerguide/msk-password.html#msk-password-limitations